### DISCUSSION TITLE
InvalidArgumentError when training the test_scan.npz

### DISCUSSION THREAD

**Discussion by TinyH** [voxelmorph/voxelmorph#345] [Q&A]
Today, when I train the VoxelMorph using test_scan.npz, in epoch 1, an error occurred, which says below. I would be really appreciated if you could help me.

Traceback (most recent call last):
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python39\lib\site-packages\tensorflow\python\client\session.py", line 1375, in _do_call
    return fn(*args)
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python39\lib\site-packages\tensorflow\python\client\session.py", line 1358, in _run_fn
    self._extend_graph()
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python39\lib\site-packages\tensorflow\python\client\session.py", line 1398, in _extend_graph
    tf_session.ExtendSession(self._session)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Node 'training/Adam/gradients/gradients/vxm_dense_transformer/map/while_grad/vxm_dense_transformer/map/while_grad': Connecting to invalid output 276 of source node vxm_dense_transformer/map/while which has 276 outputs. Try using tf.compat.v1.experimental.output_all_intermediates(True).

**Comment by zikai1** [comment]
@TinyH  Hi, I encounter the same error as follows:
**tensorflow.python.framework.errors_impl.InvalidArgumentError: Node 'training/Adam/gradients/gradients/vxm_dense_transformer/map/while_grad/vxm_dense_transformer/map/while_grad': Connecting to invalid output 154 of source node vxm_dense_transformer/map/while which has 154 outputs. Try using tf.compat.v1.experimental.output_all_intermediates(True).** How do you solve it? Thanks, TinyH.

**Comment by zikai1** [comment]
@TinyH  Hi, I have solved this problem. This is due to the tf version conflict. My solution is 
comment the 
**tf.compat.v1.disable_eager_execution()** in the **train.py** and add 
**tf.compat.v1.experimental.output_all_intermediates(True)** in it.

**Comment by TinyH** [comment]
Thank you so much. I will try your method.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年7月14日(星期四) 上午9:41
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [voxelmorph/voxelmorph] InvalidArgumentError when training the test_scan.npz (Discussion #345)





 
@TinyH  Hi, I have solved this problem. This is due to the tf version conflict. My solution is
 comment the
 **#tf.compat.v1.disable_eager_execution() ** in the train.py and add
 tf.compat.v1.experimental.output_all_intermediates(True) in it.
 
&mdash;
